### ISSUE TITLE
User Chat Layout Improvement

### ISSUE THREAD

**Issue by geanferrani123** [Ylianst/MeshCentral#596]
Guys, good morning!

What do you think about customizing the chat layout to look like this?

In my opinion, there is a huge amount of time saved from doing this on remote support, eliminating the need to open a new web browser tab.

@Ylianst, do you find this idea interesting?

In this case, the idea would be to put a layout similar to this one.

![teamviewer-chat-window](https://user-images.githubusercontent.com/55212206/67292347-45483d00-f4b9-11e9-8fcd-bf368c535e52.jpg)

**Comment by Ylianst** [comment]
This is a good idea. I will need time to add this in.

**Comment by MailYouLater** [comment]
This looks like something that will also need work on the mesh agent side.
@krayon007: Your thoughts?

**Comment by Ylianst** [comment]
Oh. I was going to do it server side since I have good control over the web application's UI. On the client side, asking @krayon007 is the right move.

**Comment by MailYouLater** [comment]
To do this right, it would include changes on both server and client side.

On the server side, it would mean moving the chat into the main MeshCentral page. Preferably in a way that's visible on any of a device's tabs (Desktop, Terminal, etc.).

On the client side, it would mean popping up a small chat interface on the desktop, preferably moveable, and set up be always on top, though those could be added after getting the basic functionality working.

**Comment by openmoto** [comment]
@Ylianst & @MailYouLater 
Would something like this work?
`window.open('/chatsessionurl','winname','directories=no,titlebar=no,toolbar=no,location=no,status=no,menubar=no,scrollbars=no,resizable=no,width=400,height=560');
`
I'm not a programmer so this may not be a good fix, but I think using a similar function to launch chat might be a quick solution.

It should launch the chat session in it's own window.

**Comment by Ylianst** [comment]
Oh wow... that worked! I will publish a new version with this.

![MC2-SmallChat](https://user-images.githubusercontent.com/1319013/67888182-e82c3700-fb09-11e9-9a27-a4b4b55fb7c9.png)

**Comment by openmoto** [comment]
Cool!
Looking forward to trying out the new version.

**Comment by Ylianst** [comment]
Published MeshCentral v0.4.3-r with this improvement. As I indicated on [Twitter](https://twitter.com/meshcentral), I will be traveling for the next 3 weeks and releases will resume on November 23rd. @krayon007 will still be actively working on improving the MeshAgent full time.

**Comment by MailYouLater** [comment]
The issue with opening it like that in a separate window, is that now any time you do any remote administration, the chat window moves behind the main window. It is an improvement over the way it worked previously, as having it in a separate window, you can place your two windows next to each other and be able to work with both, but I was hoping to have the device specific chat loaded inside the main window, similar to how the processes and services are, but when shown, it should be visible on any of that device's pages.

Oh well, perhaps it can be changed later, but it's still better than it was in it's new state.
Looking forward to seeing the future of the Agent side chat.

**Comment by Ylianst** [comment]
Agreed that was the original plan to put the chat window within the main one. However, I am out for a few weeks and this was a nice quick fix I could do quickly until I get back.

**Comment by geanferrani123** [comment]
I am also very anxious to see the news that will be implemented in Meshcentral when Ylianst returns, due to the level of evolution that the tool is having, I think the system will become something wonderful.

**Comment by Ylianst** [comment]
Thanks. FYI. Just published MeshCentral v0.4.3-s with a hidden feature. If you hold shift and press the chat button, it will open the old way with a new tab in the browser.

**Comment by openmoto** [comment]
Also in agreement with @MailYouLater about the floating window within the main window.
The current fix is great if you work with two monitors or your monitor is big enough to open side by side, at least you're still able to tell which chat is for which computer.
On the user side, it would be great to show the Admin name that is connected in place of the MeshMessenger.  
![image](https://user-images.githubusercontent.com/36439190/67930679-e7d18180-fb85-11e9-9e68-06197afbd211.png)
Perharps making it configurable would be best?
So admin can configure server to show these:
AdminName
ServerName-AdminName
DeviceGroup-AdminName

**Comment by geanferrani123** [comment]
Hello Guys!

I just tried this new chat format, I don't know if anyone else had this problem, but with the new format chat doesn't open for the remote machine, only for the technician.

According to the messages that Ylianst has posted, I believe this layout will be temporary and it will revise more calmly when returning from the trip he is making, so I'm just punctuating this to everyone's knowledge.

Has anyone else had this problem?

**Comment by Ylianst** [comment]
Allowing configuration of the top bar name in the chat box is a good idea.

**Comment by GusevVictor** [comment]
> 
> 
> This is a good idea. I will need time to add this in.

It will be very very good. Because users so stupid with open new tab in browser.

**Comment by OutbackMatt** [comment]
Also it would be good if we can set a default size / location of the chat on the Client side

**Comment by southeasterntech** [comment]
I want to +1 this.... I have frequent troubles getting the chat to open on the remote site.  It doesn't seem reliable to me on 0.6.87.  I started a chat with the remote user..... it said "waiting on user" then nothing happened.  I was watching her screen and nothing happened at all.  Ended up opening our old RMM software to shoot over a chat, not sure  why it didn't work.7.  Thx

**Comment by southeasterntech** [comment]
I think so.  Thanks for checking in

**Comment by fernandogobah** [comment]
+1
